File: src/main/java/com/dongtronic/diabot/commands/PingCommand.java
package com.dongtronic.diabot.commands;

import com.jagrosh.jdautilities.command.CommandEvent;
import com.jagrosh.jdautilities.doc.standard.CommandInfo;

import java.time.temporal.ChronoUnit;

@CommandInfo(
    name = {"Ping", "Pong"},
    description = "Checks the bot's latency"
)

public class PingCommand extends DiabotCommand {

  public PingCommand(Category category) {
    this.name = "ping";
    this.help = "checks the bot's latency";
    this.guildOnly = true;
    this.aliases = new String[]{"pong"};
    this.category = category;
  }

  @Override
  protected void execute(CommandEvent event) {
    event.reply("Ping: ...", m -> {
      long ping = event.getMessage().getCreationTime().until(m.getCreationTime(), ChronoUnit.MILLIS);
      m.editMessage("Ping: " + ping + "ms | Websocket: " + event.getJDA().getPing() + "ms").queue();
    });
  }
}

File: src/main/java/com/dongtronic/diabot/commands/PingCommand.kt
package com.dongtronic.diabot.commands

import com.jagrosh.jdautilities.command.Command
import com.jagrosh.jdautilities.command.CommandEvent
import java.time.temporal.ChronoUnit

class PingCommand(category: Command.Category) : DiabotCommand() {

    init {
        this.name = "ping"
        this.help = "checks the bot's latency"
        this.guildOnly = true
        this.aliases = arrayOf("pong")
        this.category = category
    }

    override fun execute(event: CommandEvent) {
        event.reply("Ping: ...") { m ->
            val ping = event.message.creationTime.until(m.creationTime, ChronoUnit.MILLIS)
            m.editMessage("Ping: " + ping + "ms | Websocket: " + event.jda.ping + "ms").queue()
        }
    }
}
